lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -350,6 +350,38 @@ namespace Js
     typedef FinalizableICUObject<UNumberFormat *, unum_close> FinalizableUNumberFormat;
     typedef FinalizableICUObject<UDateFormat *, udat_close> FinalizableUDateFormat;
     typedef FinalizableICUObject<UFieldPositionIterator *, ufieldpositer_close> FinalizableUFieldPositionIterator;
+    typedef FinalizableICUObject<UCollator *, ucol_close> FinalizableUCollator;
+
+    template <typename T>
+    static T *AssertProperty(DynamicObject *state, PropertyIds propertyId)
+    {
+        Var propertyValue = nullptr;
+        JavascriptOperators::GetProperty(state, propertyId, &propertyValue, state->GetScriptContext());
+
+        AssertOrFailFast(propertyValue && T::Is(propertyValue));
+
+        return T::UnsafeFromVar(propertyValue);
+    }
+
+    static int AssertIntegerProperty(DynamicObject *state, PropertyIds propertyId)
+    {
+        Var propertyValue = nullptr;
+        JavascriptOperators::GetProperty(state, propertyId, &propertyValue, state->GetScriptContext());
+
+        AssertOrFailFast(propertyValue && TaggedInt::Is(propertyValue));
```

Comment:
>TaggedInt [](start = 42, length = 9)

I could be mistaken, but I think that TaggedInt is an optimization detail that we shouldn't rely on at this level. It is possible that earlier code for some reason decided to represent this int as a tagged float or as a heap-allocated JavascriptNumber, and we shouldn't fail if it chose to do so.

Reply:
I guess we control the `toEnum` function that generates these though, so if it works then it wouldn't break due to differences in user code.

---
In reply to: [178879992](https://github.com/Microsoft/ChakraCore/pull/4926#discussion_r178879992) [](ancestors = 178879992)

Reply:
The previous version of the code here used it. I suppose that if I am going to abstract this into `AssertIntegerProperty`, it wouldn't hurt to check both TaggedInt and JavascriptNumber. I do think that allowing JavascriptNumberObjects would be too permissive though, Intl.js really shouldn't be working with Number objects.